Test/CoreSDK.Test/Shared/Extensibility/Implementation/Platform/PlatformImplementationTest.cs

```diff
@@ -45,6 +46,17 @@ public void ReadConfigurationXmlIgnoresMissingApplicationInsightsConfigurationFi
             Assert.AreEqual(0, configuration.Length);
         }
 
+        [TestMethod]
+        public void FailureToReadEnvironmentVariablesDoesNotThrowExceptions()
+        {
+            EnvironmentPermission permission = new EnvironmentPermission(EnvironmentPermissionAccess.NoAccess, "PATH");
+            permission.PermitOnly();
+            PlatformImplementation platform = new PlatformImplementation();
+            Assert.IsNull(platform.GetEnvironmentVariable("PATH"));
+            permission = null;
+            EnvironmentPermission.RevertAll();
```

Comment:
Also, validate that an error is logged into EventLog..(many tests already use TestListener to validate expected messages are logged)